Adding Led's to enhance use experience

Patel-Kamalkumar opened the pull request:
- Adding custom LED component to enhance visual representation of physical LED State which will be dipict the original system LED colour and it's behaviour.
- When user toggle the identify LED button then as current behaviour toast message will be appears for the LED status but as part of this change when user toggle identify LED button then it's appears as LED button and colour of the LED continuously blinking based on toggle button status.
- For each toggle button behaviour is different as similar as Hardware LED colour code.
For example:
- Dimms, fans, processor etc LED colour representation are amber and system identify LED colour is blue as similar as Hardware colour.

New physical LED are looks like attached images:
<img width="1164" alt="dimm slot led" src="https://github.com/ibm-openbmc/webui-vue/assets/108823339/1e533867-5c23-44e9-9574-f72636c8c241">
<img width="1728" alt="power button led" src="https://github.com/ibm-openbmc/webui-vue/assets/108823339/28c151b0-c93f-42ef-8f31-a6bdaf4ee0a2">
<img width="1585" alt="sys attention led" src="https://github.com/ibm-openbmc/webui-vue/assets/108823339/9f57f935-43f8-4258-bac2-c31962bfa35b">

mzipse commented:
If we do this, let's target this for FW1060.

gtmills commented on package.json:
```diff
@@ -17,6 +17,7 @@
   "dependencies": {
     "@carbon/icons-vue": "10.28.0",
     "@novnc/novnc": "1.2.0",
+    "@vue/runtime-dom": "3.3.4",
```
why do we need this ?

Patel-Kamalkumar replied:
sure gunner, will remove from PR.

Patel-Kamalkumar replied:
file is removed

Nikhil-Ashoka commented on package-lock.json:
We should not push the package-lock.json file. Not recommended.

Patel-Kamalkumar replied:
sure nikhil, will remove from PR.

Patel-Kamalkumar replied:
file is removed

gtmills commented:
Can you explain more on what this does? Why we need this? Can you add  a summary of the changes? Can you add a screenshot of what it looks like?

Nikhil-Ashoka commented on src/components/Global/LedComponent.vue:
```diff
@@ -0,0 +1,157 @@
+      offColours: this.offColour,
+      errColours: 'red',
+      cid: this.id,
+      //isSolidLeds: this.isSolidLed,
```
Need to remove unwanted code, console logs, debuggers.

Patel-Kamalkumar replied:
removed unwanted codes.

Nikhil-Ashoka commented on src/store/modules/HardwareStatus/FanStore.js:
```diff
@@ -29,6 +29,7 @@ const FanStore = {
           partNumber: PartNumber,
           serialNumber: SerialNumber,
           identifyLed: LocationIndicatorActive,
+          ledStatus: LocationIndicatorActive,
```
We can use identifyLed without creating ledStatus.

Patel-Kamalkumar replied:
each LED is attached with unique ledStatus, so mentioned separately.

Patel-Kamalkumar committed:
Adding Led's to enhance use experience

- Adding custom LED component to enhance visual representation of
physical LED State which will be dipict the original system LED
colour and it's behaviour.
- When user toggle the identify LED button then as current behaviour
toast message will be appears for the LED status but as part of this
change when user toggle identify LED button then it's appears as LED
button and colour of the LED continuously blinking based on toggle
button status.
- For each toggle button behaviour is different as similar as Hardware
LED colour code.
- For example:
Dimms, fans, processor etc LED colour representation are amber and
system identify LED colour is blue as similar as Hardware colour.

Signed-off-by: Kamalkumar <108823339+Patel-Kamalkumar@users.noreply.github.com>

Nikhil-Ashoka commented on src/components/Global/LedComponent.vue:
```diff
@@ -0,0 +1,133 @@
+    ledValue: { type: Boolean, default: false },
+    isSolidLed: { type: Boolean, default: false },
+    isServiceLed: { type: Boolean, default: false },
+    issysAttentionLed: { type: Boolean, default: false },
```
Change issysAttentionLed -> isSysAttentionLed.

Nikhil-Ashoka commented on src/components/Global/LedComponent.vue:
```diff
@@ -0,0 +1,133 @@
+    isSolidLed: { type: Boolean, default: false },
+    isServiceLed: { type: Boolean, default: false },
+    issysAttentionLed: { type: Boolean, default: false },
+    issysidentifyLed: { type: Boolean, default: false },
```
Change issysidentifyLed -> isSysIdentifyLed

Nikhil-Ashoka commented on src/components/Global/LedComponent.vue:
```diff
@@ -0,0 +1,133 @@
+  data() {
+    return {
+      checkbox: false,
+      blinkcolour: false,
```
blinkcolour -> **blinkColour**

Nikhil-Ashoka commented on src/store/modules/HardwareStatus/SystemStore.js:
```diff
@@ -21,11 +21,14 @@ const SystemStore = {
       system.sysAttentionLed =
         data.Oem?.IBM?.PartitionSystemAttentionIndicator ||
         data.Oem?.IBM?.PlatformSystemAttentionIndicator;
+      system.issysAttentionLed = system.sysAttentionLed;
```
issysAttentionLed  -> **isSysAttentionLed**

Nikhil-Ashoka commented on src/store/modules/HardwareStatus/SystemStore.js:
```diff
@@ -21,11 +21,14 @@ const SystemStore = {
         data.Oem?.IBM?.PlatformSystemAttentionIndicator;
+      system.issysAttentionLed = system.sysAttentionLed;
       system.locationIndicatorActive = data.LocationIndicatorActive;
+      system.issysidentifyLed = data.LocationIndicatorActive;
```
Change issysidentifyLed to **isSysIdentifyLed**

Nikhil-Ashoka commented on src/views/HardwareStatus/Inventory/InventoryServiceIndicator.vue:
```diff
@@ -135,25 +168,61 @@ export default {
     this.$store.dispatch('system/getSystem').finally(() => {
       // Emit initial data fetch complete to parent component
       this.$root.$emit('hardware-status-service-complete');
+      // this.updatePowerStatusLedState(`global.status.${this.serverStatus}`);
```
Remove comments

Nikhil-Ashoka commented on src/views/HardwareStatus/Inventory/InventoryServiceIndicator.vue:
```diff
@@ -135,25 +168,61 @@ export default {
-        .then((message) => this.successToast(message))
-        .catch(({ message }) => this.errorToast(message));
+        .then(() => {
+          //this.successToast(message);
```
I think this success Toast is required?

Nikhil-Ashoka commented on src/views/HardwareStatus/Inventory/InventoryServiceIndicator.vue:
```diff
@@ -135,25 +168,61 @@ export default {
         .dispatch('system/changeSystemAttentionLedState', systemLedState)
-        .then((message) => this.successToast(message))
-        .catch(({ message }) => this.errorToast(message));
+        .then(() => {
```
I think the success toast is still required?

Nikhil-Ashoka commented on src/views/HardwareStatus/Inventory/InventoryTableAssembly.vue:
```diff
@@ -197,6 +214,11 @@ export default {
         },
+        {
+          key: 'ledStatus',
+          label: 'Physical LED State',
```
label: '**Physical LED State**', Should be taken from the translation file.

Nikhil-Ashoka commented on src/views/HardwareStatus/Inventory/InventoryTableAssembly.vue:
```diff
@@ -286,8 +308,29 @@ export default {
         })
-        .then((message) => this.successToast(message))
-        .catch(({ message }) => this.errorToast(message));
+        .then(() => {
```
Success toast message will still be required.

Nikhil-Ashoka commented on src/views/HardwareStatus/Inventory/InventoryTableBmcManager.vue:
```diff
@@ -181,8 +196,21 @@ export default {
         })
-        .then((message) => this.successToast(message))
-        .catch(({ message }) => this.errorToast(message));
+        .then(() => {
```
Success toast will still be required.

Nikhil-Ashoka commented on src/views/HardwareStatus/Inventory/InventoryTableDimmSlot.vue:
```diff
@@ -197,6 +216,11 @@ export default {
         },
+        {
+          key: 'ledStatus',
+          label: 'Physical LED State',
```
**Physical LED State** should be taken from the translation file.

Nikhil-Ashoka commented on src/views/HardwareStatus/Inventory/InventoryTableFans.vue:
```diff
@@ -207,6 +224,11 @@ export default {
         },
+        {
+          key: 'ledStatus',
+          label: 'Physical LED State',
```
**Physical LED State** should be taken from the translation file.

Nikhil-Ashoka commented on src/views/HardwareStatus/Inventory/InventoryTablePcieSlots.vue:
```diff
@@ -139,6 +154,11 @@ export default {
         },
+        {
+          key: 'ledStatus',
+          label: 'Physical LED State',
```
Label should be taken from the translation file.

Nikhil-Ashoka commented on src/views/HardwareStatus/Inventory/InventoryTablePowerSupplies.vue:
```diff
@@ -206,6 +225,11 @@ export default {
         },
+        {
+          key: 'ledStatus',
+          label: 'Physical LED State',
```
label should be taken from the translation file

Nikhil-Ashoka commented on src/views/HardwareStatus/Inventory/InventoryTablePowerSupplies.vue:
```diff
@@ -275,13 +299,33 @@ export default {
       this.searchTotalFilteredRows = filteredItems.length;
     },
     toggleIdentifyLedValue(row) {
+      console.log('row->', row);
```
Remove console.log

Nikhil-Ashoka commented on src/views/HardwareStatus/Inventory/InventoryTableProcessors.vue:
```diff
@@ -195,6 +212,11 @@ export default {
         },
+        {
+          key: 'ledStatus',
+          label: 'Physical LED State',
```
label Should be taken from the translation file.

Nikhil-Ashoka commented on src/views/HardwareStatus/Inventory/InventoryTableProcessors.vue:
```diff
@@ -228,8 +250,25 @@ export default {
         })
-        .then((message) => this.successToast(message))
-        .catch(({ message }) => this.errorToast(message));
+        .then(() => {
```
Successtoast will still be required.

Nikhil-Ashoka commented on src/views/HardwareStatus/Inventory/InventoryFabricAdapters.vue:
```diff
@@ -194,6 +213,11 @@ export default {
         },
+        {
+          key: 'ledStatus',
+          label: 'Physical LED State',
```
Label should be taken from the translation file

Nikhil-Ashoka commented on src/views/HardwareStatus/Inventory/InventoryFabricAdapters.vue:
```diff
@@ -260,8 +284,17 @@ export default {
           memberId: row.id,
           identifyLed: row.identifyLed,
         })
-        .then((message) => this.successToast(message))
```
success toast message will still be needed

Nikhil-Ashoka commented on src/views/HardwareStatus/Inventory/InventoryTableBmcManager.vue:
```diff
@@ -151,6 +161,11 @@ export default {
         },
+        {
+          key: 'ledStatus',
+          label: 'Physical LED State',
```
label must be taken from the translation file

Nikhil-Ashoka commented on src/views/HardwareStatus/Inventory/InventoryTableChassis.vue:
```diff
@@ -134,6 +145,11 @@ export default {
         },
+        {
+          key: 'ledStatus',
+          label: 'Physical LED State',
```
label must be taken from the translation file

Nikhil-Ashoka commented on src/views/HardwareStatus/Inventory/InventoryTableChassis.vue:
```diff
@@ -157,8 +173,21 @@ export default {
           uri: row.uri,
           identifyLed: row.identifyLed,
         })
-        .then((message) => this.successToast(message))
```
success toast message would still be required.

Nikhil-Ashoka commented on src/views/HardwareStatus/Inventory/InventoryTablePowerSupplies.vue:
```diff
@@ -275,13 +299,33 @@ export default {
           uri: row.uri,
           identifyLed: row.identifyLed,
         })
-        .then((message) => this.successToast(message))
```
success toast will still be required.
